File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let ht = {};
    
    let final = [];
    
    for(let idx in nums) {
        if(!ht[target - nums[idx]]) {
            ht[nums[idx]] = idx;
        } else {
            final.push(parseInt(ht[target-nums[idx]]));
            final.push(parseInt(idx));
        }
    }
    
    return final;
    
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hashmap = {}
        
        arr = []
        
        for idx, item in enumerate(nums):
            if (target-item) not in hashmap:
                hashmap[item] = idx
            else:
                arr.append(hashmap[target-item])
                arr.append(idx)
            
        return arr